### PR TITLE
[Growth 38] As the application, I have a file helper to manipulate the template files

### PR DESCRIPTION
**Pull request by hoangmirs**
Resolves https://github.com/nimblehq/infrastructure-templates/issues/24

## What happened 👀

- Add some file helpers to copy, append, and inject content to files.
- Add a sample to add VPC modules

## Insight 📝

Parse options and pass them into our variant code.

## Proof Of Work 📹

You can test by using:
```bash
npx nimble-infra@0.2.9 generate test-infra-template
```

### CODE REVIEW

File: skeleton/aws/modules/vpc/variables.tf [hunk 1]
```diff
@@ -0,0 +1,41 @@
+variable "namespace" {
+}
```
Comment (longnd) [inline skeleton/aws/modules/vpc/variables.tf:2]: please provide the description & default type. This is the template, so the more detailed, the better, for any users of the template

```suggestion
variable "namespace" {
  description = "The namespace for the VPC"
  type = string
}
```

File: skeleton/aws/modules/vpc/variables.tf [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+variable "cidr" {
+  description = "VPC CIDR"
+  default     = "10.0.0.0/16"
+}
+
+variable "private_subnet_cidrs" {
```
Comment (longnd) [inline skeleton/aws/modules/vpc/variables.tf:9]:
```suggestion
variable "private_subnet_cidrs" {
  description = "VPC private subnet CIDRs"
```

File: skeleton/aws/modules/vpc/variables.tf [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  type    = list(any)
+  default = ["10.0.1.0/24", "10.0.2.0/24", "10.0.3.0/24"]
+}
+
+variable "public_subnet_cidrs" {
+  type    = list(any)
```
Comment (longnd) [inline skeleton/aws/modules/vpc/variables.tf:15]:
```suggestion
variable "public_subnet_cidrs" {
  description = "VPC public subnet CIDRs"
  type    = list(any)
```

File: skeleton/aws/modules/vpc/variables.tf [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  default = ["10.0.4.0/24", "10.0.5.0/24", "10.0.6.0/24"]
+}
+
+variable "enable_nat_gateway" {
+  type    = bool
```
Comment (longnd) [inline skeleton/aws/modules/vpc/variables.tf:20]:
```suggestion
variable "enable_nat_gateway" {
  description = "VPC nat gateway flag"
  type    = bool
```

File: skeleton/aws/modules/vpc/variables.tf [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  default = true
+}
+
+variable "single_nat_gateway" {
+  type    = bool
```
Comment (longnd) [inline skeleton/aws/modules/vpc/variables.tf:25]:
```suggestion
variable "single_nat_gateway" {
  description = "VPC single NAT gateway flag"
  type    = bool
```

File: skeleton/aws/modules/vpc/variables.tf [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  default = true
+}
+
+variable "one_nat_gateway_per_az" {
+  type    = bool
```
Comment (longnd) [inline skeleton/aws/modules/vpc/variables.tf:30]:
```suggestion
variable "one_nat_gateway_per_az" {
  description = "VPC one nat gateway per AZ flag"
  type    = bool
```

File: skeleton/aws/modules/vpc/variables.tf [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+  default = false
+}
+
+variable "enable_dns_hostnames" {
+  type    = bool
```
Comment (longnd) [inline skeleton/aws/modules/vpc/variables.tf:35]:
```suggestion
variable "enable_dns_hostnames" {
  description = "VPC DNS hostnames flag"
  type    = bool
```

File: skeleton/aws/modules/vpc/variables.tf [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+  default = true
+}
+
+variable "environment" {}
```
Comment (longnd) [inline skeleton/aws/modules/vpc/variables.tf:39]:
```suggestion
variable "environment" {
  description = "The environment of the app, e.g. `acme-staging`"
  type = string
}
```

File: skeleton/aws/modules/vpc/variables.tf [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+
+variable "owner" {}
```
Comment (longnd) [inline skeleton/aws/modules/vpc/variables.tf:41]:
```suggestion
variable "owner" {
  description = "The owner of the infrastructure, e.g. `acme-web`"
  type = string
}
```

File: src/commands/generate/index.ts [hunk 10]
```diff
@@ -57,14 +72,30 @@ export default class Hello extends Command {
...
-      Advanced.run()
+      console.log(options);
+
+      switch (options.infrastructureType) {
+        case 'advance':
+          Advanced.run(options);
+          break;
+        case 'basic':
+        default:
+          console.log('Not implemented this type yet');
```
Comment (longnd) [inline src/commands/generate/index.ts:94]:
```suggestion
          console.log('This type has not been implemented.');
```

**Review by byhbt — APPROVED** [review]

File: skeleton/aws/modules/vpc/variables.tf (hunk 1 above)
Comment (hoangmirs) [reply skeleton/aws/modules/vpc/variables.tf]: Added in https://github.com/nimblehq/infrastructure-templates/pull/27/commits/fe3318a48516b380ea6b34a5347ede9d6a35cfa0

File: skeleton/aws/modules/vpc/variables.tf (hunk 2 above)
Comment (hoangmirs) [reply skeleton/aws/modules/vpc/variables.tf:10]: Added in https://github.com/nimblehq/infrastructure-templates/pull/27/commits/fe3318a48516b380ea6b34a5347ede9d6a35cfa0

File: skeleton/aws/modules/vpc/variables.tf (hunk 3 above)
Comment (hoangmirs) [reply skeleton/aws/modules/vpc/variables.tf]: Added in https://github.com/nimblehq/infrastructure-templates/pull/27/commits/fe3318a48516b380ea6b34a5347ede9d6a35cfa0

File: skeleton/aws/modules/vpc/variables.tf (hunk 4 above)
Comment (hoangmirs) [reply skeleton/aws/modules/vpc/variables.tf]: Added in https://github.com/nimblehq/infrastructure-templates/pull/27/commits/fe3318a48516b380ea6b34a5347ede9d6a35cfa0

File: skeleton/aws/modules/vpc/variables.tf (hunk 5 above)
Comment (hoangmirs) [reply skeleton/aws/modules/vpc/variables.tf]: Added in https://github.com/nimblehq/infrastructure-templates/pull/27/commits/fe3318a48516b380ea6b34a5347ede9d6a35cfa0

File: skeleton/aws/modules/vpc/variables.tf (hunk 9 above)
Comment (hoangmirs) [reply skeleton/aws/modules/vpc/variables.tf]: Added in https://github.com/nimblehq/infrastructure-templates/pull/27/commits/fe3318a48516b380ea6b34a5347ede9d6a35cfa0

File: src/commands/generate/index.ts (hunk 10 above)
Comment (hoangmirs) [reply src/commands/generate/index.ts]: Fixed in https://github.com/nimblehq/infrastructure-templates/pull/27/commits/3d71b5d98ab51ef06ee7dd1d9fb47b3b474d6026

**Review by longnd — APPROVED** [review]

**Pull request merged**